usr/src/uts/common/os/zone.c

```diff
@@ -3034,7 +3034,13 @@ zone_task_rele(zone_t *zone)
 zoneid_t
 getzoneid(void)
 {
-	return (curproc->p_zone->zone_id);
+	proc_t *p = curproc;
+
+	/* This may happen for p0 during early boot, but not otherwise */
+	if (p->p_zone == NULL)
```

Comment:
Should we also check that curproc is actually p0 at this point? It's probably a bit unnecessary...

Reply:
I added an `ASSERT` in the condition, that it's `p0` which covers our assumptions.  Thanks.